src/components/PopupWithForm.js

```diff
@@ -6,22 +6,43 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._inputList = this._popupForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitBtn = this._popupElement.querySelector(".modal__button");
+    this._submitBtnText = this._submitBtn.textContent;
   }
 
   _getInputValues() {
-    const data = {};
+    const cardData = {};
     this._inputList.forEach((input) => {
-      data[input.name] = input.value;
+      cardData[input.name] = input.value;
+      cardData[input.about] = input.value;
     });
-    return data;
+    return cardData;
   }
 
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      evt.target.reset();
+      super.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well